zebra/kernel_netlink.c

```diff
@@ -997,22 +1023,8 @@ int netlink_talk_info(int (*filter)(struct nlmsghdr *, ns_id_t, int startup),
 			n->nlmsg_type, n->nlmsg_len, n->nlmsg_seq,
 			n->nlmsg_flags);
 
-	/* Send message to netlink interface. */
-	frr_with_privs(&zserv_privs) {
-		status = sendmsg(nl->sock, &msg, 0);
-		save_errno = errno;
-	}
-
-	if (IS_ZEBRA_DEBUG_KERNEL_MSGDUMP_SEND) {
-		zlog_debug("%s: >> netlink message dump [sent]", __func__);
-		zlog_hexdump(n, n->nlmsg_len);
-	}
-
-	if (status < 0) {
-		flog_err_sys(EC_LIB_SOCKET, "netlink_talk sendmsg() error: %s",
-			     safe_strerror(save_errno));
+	if (netlink_send_msg(nl, n, n->nlmsg_len) < 0)
```

Comment:
Nit: `netlink_send_msg` returns the value of `sendmsg` so it can only be `-1` on error.